core/config.c

```diff
@@ -623,6 +623,8 @@ get_config_val_other_arch(const char *var, char *val, size_t valsz, bool *app_sp
 void
 config_init(void)
 {
+    // Zero out myvals in case of reattach so that we re-load the config:
+    myvals = (const config_vals_t){0};
```

Comment:
Please put this in config_exit with the existing detach zeroing

Reply:
Done, placed it next to the existing logic which clears out `config`